Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -187,8 +187,10 @@ bool WidthIterator::hasExtraSpacing() const
 
 static void resetGlyphBuffer(GlyphBuffer& glyphBuffer, GlyphBufferStringOffset index)
 {
-    ASSERT(index < glyphBuffer.size());
-    glyphBuffer.shrink(index);
+    ASSERT(index >= 0);
```

Comment:
There is a problem with this assertion in non CG builds, as `GlyphBufferStringOffset` is defined as an `unsigned` in those platforms, so the assertion is redundant. For CG builds the type is a CFIndex, which is signed, as far as I understand.

Reply:
Maybe we switch to this then?
```
ASSERT(!std::is_signed_v<GlyphBufferStringOffset> || index >= 0);
```